adapters/tlsio_ssl_esp8266.c

```diff
@@ -0,0 +1,712 @@
+        if (ret > 0){
+            if (FD_ISSET(sock, &writeset)){
+              break;
+            }
+    
+            if (FD_ISSET(sock, &readset)){
+                memset(recv_buf, 0, recv_bytes);
+                break;
+            }
+        }else{
+            LogInfo("lwip_select returned: %d\n", ret);
+        }
+
+        retry++;
+    }
+
+    int s = sock;
+    ctx = SSL_CTX_new(TLSv1_client_method());
+    if (!ctx) {
+        LogError("create new SSL CTX failed\n");
+        return;
+    }
+    LogInfo("create new SSL CTX OK\n");
+    
+    LogInfo("SSL set fragment\n");
+    ret = SSL_set_fragment(ctx, OPENSSL_FRAGMENT_SIZE);
+
+    LogInfo("SSL new... \n");
+    ssl = SSL_new(ctx);
+
+    LogInfo("SSL set fd\n");
+    SSL_set_fd(ssl, s);
+
+    LogInfo("SSL connect... \n");
+
+    while (-1 == SSL_connect(ssl))
+    {  
+        FD_ZERO(&readset);
+        FD_SET(s, &readset);
+        FD_ZERO(&writeset);
+        FD_SET(s, &writeset);
+        FD_ZERO(&errset);
+        FD_SET(s, &errset);
+
+        ret = lwip_select(s + 1, &readset, &writeset, &errset, NULL);
+    }
+
+    LogInfo("SSL connect ok\n");
+
+    tls_io_instance->ssl = ssl;
+    tls_io_instance->ssl_context = ctx;
+}
+
+static int send_handshake_bytes(TLS_IO_INSTANCE* tls_io_instance)
+{
+    //system_print_meminfo(); // This is useful for debugging purpose.
+    //LogInfo("free heap size %d", system_get_free_heap_size()); // This is useful for debugging purpose.
+    openssl_thread_LWIP_CONNECTION(tls_io_instance);
+    tls_io_instance->tlsio_state = TLSIO_STATE_OPEN;
+    indicate_open_complete(tls_io_instance, IO_OPEN_OK);    
+    int result = 0;
+
+    return result;
+}
+
+
+static int decode_ssl_received_bytes(TLS_IO_INSTANCE* tls_io_instance)
+{
+    int result = 0;
+    unsigned char buffer[64];
+
+    int rcv_bytes = 1;
+    rcv_bytes = SSL_read(tls_io_instance->ssl, buffer, sizeof(buffer));
+    LogInfo("decode ssl recv bytes: %d\n", rcv_bytes);
+    if (rcv_bytes > 0)
+    {
+        if (tls_io_instance->on_bytes_received == NULL)
+        {
+            LogError("NULL on_bytes_received.");
+        }
+        else
+        {
+            tls_io_instance->on_bytes_received(tls_io_instance->on_bytes_received_context, buffer, rcv_bytes);
+        }
+    }
+
+    return result;
+}
+
+static void destroy_openssl_instance(TLS_IO_INSTANCE* tls_io_instance)
+{
+    if (tls_io_instance != NULL)
+    {
+        SSL_free(tls_io_instance->ssl);
+        SSL_CTX_free(tls_io_instance->ssl_context);
+    }
+}
+
+static int add_certificate_to_store(TLS_IO_INSTANCE* tls_io_instance, const char* certValue)
+{
+    return 0;
+}
+
+
+static int create_openssl_instance(TLS_IO_INSTANCE* tlsInstance)
+{
+    return 0;
+}
+
+
+/* Codes_SRS_TLSIO_SSL_ESP8266_99_005: [ The tlsio_openssl_create succeed. ]*/
+CONCRETE_IO_HANDLE tlsio_openssl_create(void* io_create_parameters)
+{
+    TLSIO_CONFIG* tls_io_config = io_create_parameters;
```

Comment:
This is derived from tlsio_openssl.h, which declares the function with `void* io_create_parameters`
```
MOCKABLE_FUNCTION(, CONCRETE_IO_HANDLE, tlsio_openssl_create, void*, io_create_parameters);
```